src/core/toga/widgets/table.py

```diff
@@ -136,3 +136,79 @@ def on_select(self, handler):
         """
         self._on_select = wrapped_handler(self, handler)
         self._impl.set_on_select(self._on_select)
+
+    def add_column(self, heading, accessor=None):
+        """
+        Add a new column to the table
+
+        :param heading:     title of the column
+        :type heading:      ``string``
+        :param accessor:    accessor of this new column
+        :type heading:      ``string``
+        """
+
+        if not accessor:
+            accessor = to_accessor(heading)
+
+        if accessor in self._accessors:
+            raise ValueError('Column name "{}" already exits'.format(accessor))
+
+        self.headings.append(heading)
+        self._accessors.append(accessor)
+
+        self._impl.add_column(heading, accessor)
+
+    def remove_column(self, heading=None, accessor=None, position=None):
+        """
+        Remove a table column
+
+        :param heading:     title of the column to remove
+        :type heading:      ``string``
+        :param accessor:    accessor of the column to remove
+        :type heading:      ``string``
+        :param position:    index of the column to remove,
+                            starting at 1
+        :type heading:      ``int``
+
+        If all parameters are sent, the priority is:
+        1) accessor
+        2) heading
+        3) position
+        """
```

Comment:
The intention here is good; however, having three arguments looks like a recipe for confusion :-)

Given that column headings could/should be translated, there's not much value in having them as an "index" for removal (since they might not actually be consistent across users). That leaves you with position and accessor - one of which is an integer. That means you can do a type check to work out whether the argument is an index or an accessor, and raise an error if the index is invalid or the accessor doesn't exist.

Reply:
I fixed. I agree that is better this way 👍